node-src/index.ts

```diff
@@ -102,17 +109,26 @@ export async function run({
   };
 }
 
-export async function runAll(ctx) {
+// Entry point for testing
+export async function runAll(ctx: Context) {
+  setExitCode(ctx, exitCodes.OK);
+
+  ctx.http = (ctx.http as HTTPClient) || new HTTPClient(ctx);
+
   // Run these in parallel; neither should ever reject
   await Promise.all([runBuild(ctx), checkForUpdates(ctx)]);
 
   if (ctx.exitCode === 0 || ctx.exitCode === 1) {
     await checkPackageJson(ctx);
   }
 
-  if (ctx.options.diagnostics) {
+  if (ctx.flags?.diagnostics || ctx.extraOptions?.diagnostics) {
```

Comment:
`runBuild` should have set `ctx.options` from `flags`/`extraOptions`, so this change is redundant. But I don't love the indirection.

Reply:
`runBuild` may throw before it even sets `ctx.options` (e.g. when parsing a configuration file or flags). This is further addressed in [this PR](https://github.com/chromaui/chromatic-cli/pull/840).